x-pack/filebeat/input/s3/input.go

```diff
@@ -181,7 +181,17 @@ func (p *s3Input) Run() {
 
 		awsConfig := p.awsConfig.Copy()
 		awsConfig.Region = regionName
+
+		// check if endpoint is given from configuration
+		if p.config.AwsConfig.Endpoint != "" {
```

Comment:
Wondering if this functionality could be somehow abstracted into a utility function and being called from all the different points with the proper parameters 🤔 . This function could receive the configs and thetype, do the check if endpoint is et and set it accordingly. wdyt?

Something like:
```
awsConfig = EnrichConfigWithEndpoint(awsConfig, type, p.config.AwsConfig)
```

Reply:
Good point!! I can create a function under libbeat/common and can be used for both FB and MB! Thanks!

Reply:
done!